File: frontend/src/app/features/models/post.ts
import { PostMedia } from "./postMedia";
import { PostLocation } from "./location";
export interface Post{
    id: string;
    userId: string;
    content?: string;
    media: PostMedia[];
    createdAt: Date;
    likedBy: string[];
    commentsCount:number;
    location?: PostLocation;
}

File: backend/ImigramAPI/ImigramAPI/Models/Post.cs
using MongoDB.Bson.Serialization.Attributes;

namespace ImigramAPI.Models
{
    public class Post
    {
        [BsonId]
        [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
        public string Id { get; set; }
        public string UserId { get; set; }
        public string? Content { get; set; }
        public List<PostMedia> Media { get; set; } = new List<PostMedia>();
        public DateTime CreatedAt { get; set; }
        public List<string> LikedBy { get; set; } = new List<string>();
        public int CommentsCount { get; set; }
        public PostLocation? Location { get; set; }
        public bool IsDeleted { get; set; }
    }
}
